telegram/ext/jobqueue.py

```diff
@@ -63,12 +61,36 @@ def __init__(self):
         else:
             self._dispatcher = None
         self.logger = logging.getLogger(self.__class__.__name__)
-        self.__start_lock = Lock()
-        self.__next_peek_lock = Lock()  # to protect self._next_peek & self.__tick
-        self.__tick = Event()
-        self.__thread = None
-        self._next_peek = None
-        self._running = False
+        self.scheduler = BackgroundScheduler(timezone=pytz.utc)
+        self.scheduler.add_listener(self._update_persistence,
+                                    mask=EVENT_JOB_EXECUTED | EVENT_JOB_ERROR)
+
+    def _build_args(self, job):
+        if self._dispatcher.use_context:
+            return [CallbackContext.from_job(job, self._dispatcher)]
+        return [self._dispatcher.bot, job]
+
+    def _tz_now(self):
+        return datetime.datetime.now(self.scheduler.timezone)
+
+    def _update_persistence(self, event):
+        self._dispatcher.update_persistence()
+
+    def _parse_time_input(self, time, shift_day=False):
+        if time is None:
+            return None
+        if isinstance(time, (int, float)):
+            return self._tz_now() + datetime.timedelta(seconds=time)
+        if isinstance(time, datetime.timedelta):
+            return self._tz_now() + time
+        if isinstance(time, datetime.time):
+            dt = datetime.datetime.combine(datetime.datetime.now().date(), time,
+                                           tzinfo=time.tzinfo or self.scheduler.timezone)
```

Comment:
I don't know about this... Are we dropping 3.5 support (or did we already)? The third argument to `combine` has been added in 3.6.

Apart from that, couldn't `datetime.datetime.now().date()` return the wrong date this way?

Reply:
Uh, you're right. Couldn't tell, because TG messed up our tests again …. But we're dropping py3.5 an #1920 anyway, so we might as well do here.

And right once more. I'll add it back.